domains/client/block-preprocessor/src/lib.rs

```diff
@@ -501,6 +501,11 @@ where
         let bundle_estimated_weight = bundle.estimated_weight();
         let mut maybe_invalid_bundle_type = None;
 
+        // Note: It is okay to use stateless here since all the bundle checks currently do not
+        // require state. But of any checks in the future requires a state read that was part of the
+        // genesis ex: `SelfChainId`, stateless runtime will panic.
+        // So ideal to set the genesis storage as fraud proof already have access to that and would be
+        // no different in terms of verification on both the sides
```

Comment:
If block-preprocessor and FP verifier use genesis storage to do the bundle checks, then the bundle producer also needs to do the same check...

As long as we have tests cover all the bundle checks, if we somehow accidentally query the state during these bundle checks, then we should have the test fail and detect it thus should not trigger in live network siliently.